### ISSUE TITLE
Allow returning a relation from the calculated_attribute lambda

### ISSUE THREAD

**Issue by lowjoel** [aha-app/calculated_attributes#1]
In lib/calculated_attributes.rb:76, what if the lambda returns a relation? I am thinking of constructing the query using the squeel DSL, then returning that to the library.

There's currently a check for strings, for raw Arel, how about an AR::Relation, which can easily be .to_sql'ed? (or even better, do a conversion to actual Arel?)

**Comment by schneidmaster** [comment]
Can you provide an example of the squeel query you want to use in the lambda? I'm not sure I understand when this would be necessary.

**Comment by lowjoel** [comment]
Strictly speaking, it isn't necessary. I can always use `to_sql`. I figure that using a Squeel query removes some of the verbosity of a) writing raw SQL as well as b) writing an Arel query. Squeel operates on Arel through an AR intermediate.

As an example:
```ruby
  calculated :submitted_at, (lambda do
    Course::Assessment::Answer.where do
      course_assessment_answers.submission_id == course_assessment_submissions.id
    end.select { max(course_assessment_answers.submitted_at) }.to_sql
  end)
```

Squeel returns an AR relation, so in theory it can be chained. That might be useful in constructing a more complex query.

**Comment by schneidmaster** [comment]
That PR should do what you want, right?

**Comment by lowjoel** [comment]
Cool! I'll give it a shot later. 

There's another issue after #2 and #3. I'll open a ticket later. I think I'll help to write a spec for it.

**Comment by lowjoel** [comment]
Also, maybe there should be a line in README.md about the use of things that respond to `to_sql`
